<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>02_clean</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="02_clean_files/libs/clipboard/clipboard.min.js"></script>
<script src="02_clean_files/libs/quarto-html/quarto.js"></script>
<script src="02_clean_files/libs/quarto-html/popper.min.js"></script>
<script src="02_clean_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="02_clean_files/libs/quarto-html/anchor.min.js"></script>
<link href="02_clean_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="02_clean_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="02_clean_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="02_clean_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="02_clean_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">02_clean</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(stringr)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyr)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"../R/99_proj_func.R"</span>)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="co">#mouse datasets</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>mouse_data <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="at">file =</span> <span class="st">"../data/01_dat_load_mouse.tsv.gz"</span>)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>mouse_suppl_data <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="at">file =</span> <span class="st">"../data/01_dat_load_mouse_suppl.tsv.gz"</span>)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="co">#human_datasets</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>human_data <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="at">file =</span> <span class="st">"../data/01_dat_load_human.tsv.gz"</span>)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>human_suppl_data <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="at">file =</span> <span class="st">"../data/01_dat_load_human_suppl.tsv.gz"</span>)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>data_list <span class="ot">&lt;-</span> <span class="fu">list</span>(mouse_data, mouse_suppl_data,human_data, human_suppl_data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>First we need to see how the data are represented</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 14
   gene_id    symbol `DMSO-1` `DMSO-2` `DMSO-3` `G007LK-1` `G007LK-2` `G007LK-3`
   &lt;chr&gt;      &lt;chr&gt;     &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt;      &lt;dbl&gt;      &lt;dbl&gt;      &lt;dbl&gt;
 1 ENSMUSG00… GNAI3    75.0    76.6       54.9       74.9       70.0       59.2 
 2 ENSMUSG00… PBSN      0       0          0          0          0          0   
 3 ENSMUSG00… CDC45    48.8    55.6       49.9       50.4       57.7       49.6 
 4 ENSMUSG00… SCML2     1.20    2.81       3.15       1.39       2.50       4.34
 5 ENSMUSG00… APOH      0.178   0.0621     0          0          0          0   
 6 ENSMUSG00… NARF      8.30    8.14      14.5       11.0        8.86      22.1 
 7 ENSMUSG00… CAV2    141.    112.        77.5      142.       107.        89.8 
 8 ENSMUSG00… KLF6     45.8    37.8       45.5       48.4       36.3       45.4 
 9 ENSMUSG00… SCMH1    23.1    17.4       19.4       20.3       20.3       17.2 
10 ENSMUSG00… COX5A   493.    518.       452.       488.       515.       436.  
# ℹ 6 more variables: `WNT-1` &lt;dbl&gt;, `WNT-2` &lt;dbl&gt;, `WNT-3` &lt;dbl&gt;,
#   `WNT_G007LK-1` &lt;dbl&gt;, `WNT_G007LK-2` &lt;dbl&gt;, `WNT_G007LK-3` &lt;dbl&gt;</code></pre>
</div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 38
   gene_id     symbol A375.DMSO A375.G007.LK FemxI.DMSO FemxI.G007.LK FemxV.DMSO
   &lt;chr&gt;       &lt;chr&gt;      &lt;dbl&gt;        &lt;dbl&gt;      &lt;dbl&gt;         &lt;dbl&gt;      &lt;dbl&gt;
 1 ENSG000000… TSPAN6    15.6         15.6      21.8         22.2         22.7  
 2 ENSG000000… TNMD       0            0         0            0            0    
 3 ENSG000000… DPM1     150.         169.       88.1         87.1         62.1  
 4 ENSG000000… SCYL3      7.09         6.04      4.69         5.65         3.79 
 5 ENSG000000… C1orf…    24.1         22.6      24.5         26.8         22.8  
 6 ENSG000000… FGR        0            0         0            0.0102       0    
 7 ENSG000000… CFH        0.967        0.776     0.0366       0.00739      0.128
 8 ENSG000000… FUCA2     24.7         25.4      38.9         38.8         43.9  
 9 ENSG000000… GCLC      34.2         24.1      23.8         29.8         29.0  
10 ENSG000000… NFYA      20.0         16.4      19.8         20.1         19.7  
# ℹ 31 more variables: FemxV.G007.LK &lt;dbl&gt;, LOX.DMSO &lt;dbl&gt;, LOX.G007.LK &lt;dbl&gt;,
#   MeWo.DMSO &lt;dbl&gt;, MeWo.G007.LK &lt;dbl&gt;, Skmel28.DMSO &lt;dbl&gt;,
#   Skmel28.G007.LK &lt;dbl&gt;, WM115.DMSO &lt;dbl&gt;, WM115.G007.LK &lt;dbl&gt;,
#   WM1341.DMSO &lt;dbl&gt;, WM1341.G007.LK &lt;dbl&gt;, WM1366.DMSO &lt;dbl&gt;,
#   WM1366.G007.LK &lt;dbl&gt;, WM1382.DMSO &lt;dbl&gt;, WM1382.G007.LK &lt;dbl&gt;,
#   WM239.DMSO &lt;dbl&gt;, WM239.G007.LK &lt;dbl&gt;, WM266.DMSO &lt;dbl&gt;,
#   WM266.G007.LK &lt;dbl&gt;, WM35.DMSO &lt;dbl&gt;, WM35.G007.LK &lt;dbl&gt;, …</code></pre>
</div>
</div>
<section id="check-for-the-possible-nas-in-the-4-data-sets" class="level3">
<h3 class="anchored" data-anchor-id="check-for-the-possible-nas-in-the-4-data-sets">Check for the possible NAs in the 4 data sets</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">walk</span>(data_list, nas_present)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>There are no NA values in the data frame.
There are no NA values in the data frame.
There are no NA values in the data frame.
There are no NA values in the data frame.</code></pre>
</div>
</div>
<p>First of all we observe that none of the data sets contain any missing values</p>
</section>
<section id="mouse-data-set-description" class="level3">
<h3 class="anchored" data-anchor-id="mouse-data-set-description"><strong>Mouse data set description:</strong></h3>
<p>Looking at this data set we can see that it represents the experimental compounds as variables and lists genes by their assembly abbreviations and their common name as rows. Each cell in the data set holds the expression levels of individual genes within the B16-F10 cell line. To optimize the utilization of this data set we can perform a tidy procedure.</p>
</section>
<section id="human-data-set-description" class="level3">
<h3 class="anchored" data-anchor-id="human-data-set-description"><strong>Human data set description:</strong></h3>
<p>This data set comprises of a gene id, a column containing common gene name and multiple columns representing the cell lines that were used for the experiment combined with their treatment status. This data set also requires some cleaning before analyzing it.</p>
</section>
<section id="mouse-data-sets" class="level1">
<h1>Mouse data sets</h1>
<section id="mouse-data-cleaning" class="level3">
<h3 class="anchored" data-anchor-id="mouse-data-cleaning">Mouse data cleaning</h3>
<p>Converting to long format</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to long format </span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>mouse_data_long <span class="ot">&lt;-</span> mouse_data <span class="sc">|&gt;</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">cols =</span> <span class="sc">-</span><span class="fu">c</span>(gene_id, symbol),</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">names_to =</span> <span class="st">"treatment_replicate"</span>,</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">values_to =</span> <span class="st">"expression_levels"</span></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Separate the treatment and replicate information</span></span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>mouse_data_separated_repl <span class="ot">&lt;-</span> mouse_data_long <span class="sc">|&gt;</span></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(</span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">col =</span> treatment_replicate,</span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">into =</span> <span class="fu">c</span>(<span class="st">"treatment"</span>, <span class="st">"replicate"</span>),</span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">sep =</span> <span class="st">"-"</span></span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a>    )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>mouse_clean <span class="ot">&lt;-</span> mouse_data_separated_repl <span class="sc">|&gt;</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="co"># The symbol variable does not provide us with addtional information so we are going to drop it</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>symbol) <span class="sc">|&gt;</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> gene_id, </span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>              <span class="at">values_from =</span> expression_levels</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>              )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 12 × 10
   treatment  replicate ENSMUSG00000000001 ENSMUSG00000000003 ENSMUSG00000000028
   &lt;chr&gt;      &lt;chr&gt;                  &lt;dbl&gt;              &lt;dbl&gt;              &lt;dbl&gt;
 1 DMSO       1                       75.0                  0               48.8
 2 DMSO       2                       76.6                  0               55.6
 3 DMSO       3                       54.9                  0               49.9
 4 G007LK     1                       74.9                  0               50.4
 5 G007LK     2                       70.0                  0               57.7
 6 G007LK     3                       59.2                  0               49.6
 7 WNT        1                       74.9                  0               49.5
 8 WNT        2                       73.4                  0               51.4
 9 WNT        3                       57.3                  0               47.4
10 WNT_G007LK 1                       74.1                  0               47.9
11 WNT_G007LK 2                       71.7                  0               53.1
12 WNT_G007LK 3                       68.3                  0               44.2
# ℹ 5 more variables: ENSMUSG00000000037 &lt;dbl&gt;, ENSMUSG00000000049 &lt;dbl&gt;,
#   ENSMUSG00000000056 &lt;dbl&gt;, ENSMUSG00000000058 &lt;dbl&gt;,
#   ENSMUSG00000000078 &lt;dbl&gt;</code></pre>
</div>
</div>
<p>After the cleaning process the resulting format presents various treatments as rows with replicate number and gene_ids as variables.</p>
</section>
<section id="supplementary-mouse-data-cleaning" class="level3">
<h3 class="anchored" data-anchor-id="supplementary-mouse-data-cleaning">Supplementary mouse data cleaning</h3>
<p>Upon inspection of the data set it is seen that most supplementary data consists of metadata concerning the experiment that do not vary between the individual drug experiments.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Clean the column names from the mouse supplementary data set</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>mouse_suppl_clean <span class="ot">&lt;-</span>mouse_suppl_data <span class="sc">|&gt;</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>  janitor<span class="sc">::</span><span class="fu">clean_names</span>() <span class="sc">|&gt;</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename_with</span>(</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>    <span class="sc">~</span> <span class="fu">str_remove</span>(., <span class="st">"characteristics_"</span>),<span class="co"># remove the characteristics_ from the column names</span></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">starts_with</span>(<span class="st">"characteristics_"</span>),</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename_with</span>(</span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>       <span class="sc">~</span><span class="fu">str_remove</span>(., <span class="st">"comment_"</span>), <span class="co">#remove the comment_ from the column names</span></span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">starts_with</span>(<span class="st">"comment_"</span>)</span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Some extra cleaning with the variable factor_value_compound</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>mouse_suppl_clean <span class="ot">&lt;-</span> mouse_suppl_clean <span class="sc">|&gt;</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">factor_value_compound=</span> <span class="fu">case_when</span>(</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>      factor_value_compound <span class="sc">==</span> <span class="st">"dimethyl sulfoxide (0.1 percent)"</span> <span class="sc">~</span> <span class="st">"0.1%"</span>,</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>      factor_value_compound <span class="sc">==</span> <span class="st">"G007LK (1 micromolar)"</span> <span class="sc">~</span> <span class="st">"1μM"</span>,</span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>      factor_value_compound <span class="sc">==</span> <span class="st">"WNT3a (0.5 microgram per milliliter)"</span> <span class="sc">~</span> <span class="st">"0.5mg/ml "</span>,</span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>      factor_value_compound <span class="sc">==</span> <span class="st">"WNT3a (0.5 microgram per milliliter); G007LK (1 micromolar)"</span> <span class="sc">~</span> <span class="st">"0.5mg/ml; 1μΜ"</span>,</span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>      <span class="cn">TRUE</span> <span class="sc">~</span> factor_value_compound) <span class="co"># keep the original value of the compound_dose column if the conditions are not met</span></span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>    )<span class="sc">|&gt;</span></span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">compound_dose =</span> factor_value_compound) <span class="sc">|&gt;</span></span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a>  <span class="co">#get rid of the redundant columns</span></span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(extract_name, assay_name)) <span class="co">#those two columns contain the same information as the source_name</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 32
   source_name ena_sample bio_sd_sample organism disease organism_part cell_line
   &lt;chr&gt;       &lt;chr&gt;      &lt;chr&gt;         &lt;chr&gt;    &lt;chr&gt;   &lt;chr&gt;         &lt;chr&gt;    
 1 DMSO-1      ERS3548927 SAMEA5745642  Mus mus… melano… skin          B16-F10 …
 2 DMSO-1      ERS3548927 SAMEA5745642  Mus mus… melano… skin          B16-F10 …
 3 DMSO-2      ERS3548928 SAMEA5745643  Mus mus… melano… skin          B16-F10 …
 4 DMSO-2      ERS3548928 SAMEA5745643  Mus mus… melano… skin          B16-F10 …
 5 DMSO-3      ERS3548929 SAMEA5745644  Mus mus… melano… skin          B16-F10 …
 6 DMSO-3      ERS3548929 SAMEA5745644  Mus mus… melano… skin          B16-F10 …
 7 G007LK-1    ERS3548930 SAMEA5745645  Mus mus… melano… skin          B16-F10 …
 8 G007LK-1    ERS3548930 SAMEA5745645  Mus mus… melano… skin          B16-F10 …
 9 G007LK-2    ERS3548931 SAMEA5745646  Mus mus… melano… skin          B16-F10 …
10 G007LK-2    ERS3548931 SAMEA5745646  Mus mus… melano… skin          B16-F10 …
# ℹ 25 more variables: provider &lt;chr&gt;, material_type &lt;chr&gt;,
#   protocol_ref_10 &lt;chr&gt;, protocol_ref_11 &lt;chr&gt;, protocol_ref_12 &lt;chr&gt;,
#   performer_13 &lt;chr&gt;, library_layout &lt;chr&gt;, library_selection &lt;chr&gt;,
#   library_source &lt;chr&gt;, library_strategy &lt;chr&gt;, nominal_length &lt;dbl&gt;,
#   nominal_sdev &lt;dbl&gt;, protocol_ref_21 &lt;chr&gt;, performer_22 &lt;chr&gt;,
#   technology_type &lt;chr&gt;, ena_experiment &lt;chr&gt;, scan_name &lt;chr&gt;,
#   submitted_file_name &lt;chr&gt;, ena_run &lt;chr&gt;, fastq_uri &lt;chr&gt;, …</code></pre>
</div>
</div>
<p>That is it for the cleaning process of the mouse data and supplementary mouse data.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># # Write the two clean data sets to tsv files</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(<span class="at">x =</span> mouse_clean, </span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>          <span class="at">file =</span> (<span class="st">"../data/02_dat_clean_mouse.tsv.gz"</span>))</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(<span class="at">x =</span> mouse_suppl_clean,</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>          <span class="at">file =</span> (<span class="st">"../data/02_dat_clean_mouse_suppl.tsv.gz"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="human-data-sets" class="level2">
<h2 class="anchored" data-anchor-id="human-data-sets">Human data sets</h2>
<p>This data set is also untidy. Therefore we are going to transform it via pivoting longer. But before proceeding, it is evident that the column names contain two pieces of information, the cell line and the compound used in each experiment. Therefore we have to split these two.</p>
<p>This is our approach of doing that:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>human_trans <span class="ot">&lt;-</span>human_data <span class="sc">|&gt;</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename_with</span>( </span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>    <span class="sc">~</span><span class="fu">case_when</span>( </span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>      <span class="co">#We see that the columns containing the cell lines contain the name of the cell lines and the DMSO or G007.LK compounds separated by a "."</span></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>       <span class="fu">str_detect</span>(., <span class="st">".DMSO"</span>) <span class="sc">~</span> <span class="fu">str_replace_all</span>(.,</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>                                             <span class="st">".DMSO"</span>,</span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>                                             <span class="st">"_DMSO"</span>),</span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>       <span class="fu">str_detect</span>(., <span class="st">".G007.LK"</span>) <span class="sc">~</span> <span class="fu">str_replace_all</span>(.,</span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>                                                <span class="st">".G007.LK"</span>,</span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>                                                <span class="st">"_G007LK"</span>),</span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a>       <span class="cn">TRUE</span> <span class="sc">~</span> .  <span class="co"># keep the original name of the columns if the aforementioned conditions are not met</span></span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a>     )</span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a>   ) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Pivot the human dataframe</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>human_clean_long <span class="ot">&lt;-</span> human_trans <span class="sc">|&gt;</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">cols =</span> <span class="sc">-</span>(gene_id<span class="sc">:</span>symbol),</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">names_to =</span> <span class="fu">c</span>(<span class="st">"cell_line"</span>,</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>                 <span class="st">"treatment"</span>),  </span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">values_to =</span> <span class="st">"expression_levels"</span>,</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">names_sep =</span> <span class="st">"_"</span></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>      )<span class="sc">|&gt;</span></span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>symbol) <span class="co"># we exclude the symbol for the same reason we did when we were cleaning the mouse dataset</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Now we are going to finish our cleaning process by getting back to wide format but with the different genes as variables in the column names</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>human_clean <span class="ot">&lt;-</span> human_clean_long <span class="sc">|&gt;</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">names_from =</span> gene_id,</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">values_from =</span> expression_levels</span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 39,077
   cell_line treatment ENSG00000000003 ENSG00000000005 ENSG00000000419
   &lt;chr&gt;     &lt;chr&gt;               &lt;dbl&gt;           &lt;dbl&gt;           &lt;dbl&gt;
 1 A375      DMSO                 15.6               0           150. 
 2 A375      G007LK               15.6               0           169. 
 3 FemxI     DMSO                 21.8               0            88.1
 4 FemxI     G007LK               22.2               0            87.1
 5 FemxV     DMSO                 22.7               0            62.1
 6 FemxV     G007LK               22.8               0            76.3
 7 LOX       DMSO                 21.6               0            62.9
 8 LOX       G007LK               23.6               0            71.1
 9 MeWo      DMSO                 12.3               0           104. 
10 MeWo      G007LK               12.8               0            93.9
# ℹ 39,072 more variables: ENSG00000000457 &lt;dbl&gt;, ENSG00000000460 &lt;dbl&gt;,
#   ENSG00000000938 &lt;dbl&gt;, ENSG00000000971 &lt;dbl&gt;, ENSG00000001036 &lt;dbl&gt;,
#   ENSG00000001084 &lt;dbl&gt;, ENSG00000001167 &lt;dbl&gt;, ENSG00000001460 &lt;dbl&gt;,
#   ENSG00000001461 &lt;dbl&gt;, ENSG00000001497 &lt;dbl&gt;, ENSG00000001561 &lt;dbl&gt;,
#   ENSG00000001617 &lt;dbl&gt;, ENSG00000001626 &lt;dbl&gt;, ENSG00000001629 &lt;dbl&gt;,
#   ENSG00000001630 &lt;dbl&gt;, ENSG00000001631 &lt;dbl&gt;, ENSG00000002016 &lt;dbl&gt;,
#   ENSG00000002079 &lt;dbl&gt;, ENSG00000002330 &lt;dbl&gt;, ENSG00000002549 &lt;dbl&gt;, …</code></pre>
</div>
</div>
<p>Now we are going to clean the supplementary human file.</p>
<p>The source name column has the same format as the columns we pivoted before and separated them into cell_line and compound columns.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Clean the column names from the human supplementary data</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>human_suppl_clean <span class="ot">&lt;-</span> human_suppl_data <span class="sc">|&gt;</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>  janitor<span class="sc">::</span><span class="fu">clean_names</span>() <span class="sc">|&gt;</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename_with</span>(</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>    <span class="sc">~</span> <span class="fu">str_remove</span>(., <span class="st">"characteristics_"</span>),</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">starts_with</span>(<span class="st">"characteristics_"</span>)</span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>   <span class="fu">rename_with</span>(</span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>       <span class="sc">~</span><span class="fu">str_remove</span>(., <span class="st">"comment_"</span>),</span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">starts_with</span>(<span class="st">"comment_"</span>)</span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>    )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co"># We separate the first column</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="co"># We notice an inconsistency with the names of the cell lines which creates some problems when we try to separate the two components of the first column</span></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>human_suppl_clean_change_treat <span class="ot">&lt;-</span> human_suppl_clean <span class="sc">|&gt;</span></span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">source_name =</span></span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">case_when</span>(</span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>      <span class="co">#We see that the column source name contains the name of the cell lines and the compounds separated by a "."</span></span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>       <span class="fu">str_detect</span>(source_name, <span class="st">"</span><span class="sc">\\</span><span class="st">.DMSO$"</span>) <span class="sc">~</span> <span class="fu">str_replace</span>(source_name,</span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>                                                         <span class="st">"</span><span class="sc">\\</span><span class="st">.DMSO"</span>,</span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>                                                         <span class="st">"_DMSO"</span>),</span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>       <span class="fu">str_detect</span>(source_name, <span class="st">"</span><span class="sc">\\</span><span class="st">.G007.LK"</span>) <span class="sc">~</span> <span class="fu">str_replace</span>(source_name,</span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>                                                           <span class="st">"</span><span class="sc">\\</span><span class="st">.G007.LK"</span>,</span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a>                                                           <span class="st">"_G007LK"</span>),</span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a>       <span class="cn">TRUE</span> <span class="sc">~</span> source_name  <span class="co"># Keep the original name of the columns if the aforementioned conditions are not met</span></span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a>     )<span class="co"># </span></span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a>   ) <span class="sc">|&gt;</span></span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(source_name,</span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a>           <span class="at">into =</span> <span class="fu">c</span>(<span class="st">"cell_line"</span>, <span class="st">"treatment"</span>),</span>
<span id="cb18-19"><a href="#cb18-19" aria-hidden="true" tabindex="-1"></a>           <span class="at">sep =</span> <span class="st">"</span><span class="sc">\\</span><span class="st">_"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>human_suppl_comp_dose <span class="ot">&lt;-</span> human_suppl_clean_change_treat <span class="sc">|&gt;</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">Compound_dose =</span> <span class="fu">str_c</span>(factor_value_dose ,unit_dose_unit, <span class="at">sep =</span> <span class="st">" "</span>),</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">Compound_dose =</span> <span class="fu">case_when</span>(</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>    Compound_dose <span class="sc">==</span> <span class="st">"0.01 percent"</span> <span class="sc">~</span> <span class="st">"0.01%"</span>,</span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>    Compound_dose <span class="sc">==</span> <span class="st">"1 micromolar"</span> <span class="sc">~</span> <span class="st">"1μM"</span>,</span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> Compound_dose</span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>      ) <span class="co"># Keeps the original value of the column if the condition is not met</span></span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>  )<span class="sc">|&gt;</span> <span class="fu">select</span>(</span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>      <span class="sc">-</span><span class="fu">c</span>(factor_value_dose,unit_dose_unit)</span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>    )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Extract the redundant columns</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>human_suppl_clean_final <span class="ot">&lt;-</span> human_suppl_comp_dose <span class="sc">|&gt;</span></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(extract_name,</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>            assay_name,</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>            factor_value_compound))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 24
   cell_line treatment organism     developmental_stage disease  genotype
   &lt;chr&gt;     &lt;chr&gt;     &lt;chr&gt;        &lt;chr&gt;               &lt;chr&gt;    &lt;chr&gt;   
 1 A375      DMSO      Homo sapiens adult               melanoma BRAF    
 2 LOX       DMSO      Homo sapiens adult               melanoma BRAF    
 3 Skmel28   DMSO      Homo sapiens adult               melanoma BRAF    
 4 WM115     DMSO      Homo sapiens adult               melanoma BRAF    
 5 WM1341    DMSO      Homo sapiens adult               melanoma BRAF    
 6 WM239     DMSO      Homo sapiens adult               melanoma BRAF    
 7 WM266     DMSO      Homo sapiens adult               melanoma BRAF    
 8 WM35      DMSO      Homo sapiens adult               melanoma BRAF    
 9 WM45.1    DMSO      Homo sapiens adult               melanoma BRAF    
10 WM793     DMSO      Homo sapiens adult               melanoma BRAF    
# ℹ 18 more variables: organism_part &lt;chr&gt;, material_type &lt;chr&gt;,
#   protocol_ref_9 &lt;chr&gt;, protocol_ref_10 &lt;chr&gt;, protocol_ref_11 &lt;chr&gt;,
#   performer_12 &lt;chr&gt;, library_layout &lt;chr&gt;, library_selection &lt;chr&gt;,
#   library_source &lt;chr&gt;, library_strategy &lt;chr&gt;, protocol_ref_18 &lt;chr&gt;,
#   performer_19 &lt;chr&gt;, technology_type &lt;chr&gt;, protocol_ref_22 &lt;chr&gt;,
#   performer_23 &lt;chr&gt;, derived_array_data_file &lt;chr&gt;,
#   derived_array_express_ftp_file &lt;chr&gt;, Compound_dose &lt;chr&gt;</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Write the two clean datasets to tsv files</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(<span class="at">x =</span> human_clean,</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>          <span class="at">file =</span> <span class="st">"../data/02_dat_clean_human.tsv.gz"</span>)</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(<span class="at">x =</span> human_suppl_clean_final,</span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>          <span class="at">file =</span> <span class="st">"../data/02_dat_clean_human_suppl.tsv.gz"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>