---
title: "Human augmentation"
format: html
editor: visual
---

```{r}
library(tidyverse)
library(dplyr)
```

```{r}
human_data_clean_long
```

```{r}
# Now we are going to make our data wide making the values of the "Genes" our variables  since we want to see how are they affected by the compound G007.LK
human_data_clean_wide <- human_data_clean_long |> 
  pivot_wider(
    names_from = Gene,
    values_from = Expression_levels
  )

human_data_clean_wide
```

We are going to merge our clean human data sets and select only the variables that we want

```{r}
human_joined <- human_data_clean_long |> 
  left_join(human_suppl_clean, join_by(Treatment, Cell_line)) |>
  select(Gene,
         Cell_line,
         Treatment,
         Expression_levels,
         Genotype) |>
  relocate(Genotype, .before = 2)

human_joined
```

Thinkg to Keep in mind. Our data have been already normalized with the TPM method when the researchers used the sleuth v0.29 in their experiment.

\##############################################################################

### This is were we create the new metrics for our augmented file

In order to work wit our DMSO and G007.LK values we chose to move on with log transformation.

Because we can't log on zeros we add on all the values the 0.99% of the minimum positive value as an offset

```{r}
min_pos_value <- human_joined |>
  select(Expression_levels) |>
  filter(Expression_levels != 0) |>
  summarize(min = min(Expression_levels)) |>
  pull()


human_joined_aug <- human_joined |> 
  mutate(Log_expression_levels = log(Expression_levels + 0.99*min_pos_value)
         ) |> 
  select(-Expression_levels)

human_joined_aug
```

```{r}
# We also want to create a wide version where the compounds are on different columns
human_joined_aug_wide <- human_joined_aug |> 
  pivot_wider(
    names_from = Treatment,
    values_from = Log_expression_levels
  )

human_joined_aug_wide
```

```{r}
human_joined_aug_wide <- human_joined_aug_wide |>
  mutate(log2fold = G007.LK / DMSO)

human_joined_aug_wide
```

We want to create a subset where we don't take into account genes with the same expression levels with and without treatment.

```{r}
human_diff_expr <- human_joined_aug_wide |>
  filter(log2fold != 1)

human_diff_expr
```

From that illustration we can see that there is not a global up regulation or down regulation for all genes.

We can perform several test to assess the statistical analysis of this result

\##########################################

Alternative approaches in order to get rid of the 0s####

\############################################

1st approach: The most naive one

But before we do that we have to log transform our data and add a constant (e.g. 1)

```{r}
#log_transformation +1 to deal with the huge number of zeros

#we do this before the tests
human_data_transform1<- human_data_transform1 |> 
  mutate(
    log2_DMSO = log2(DMSO +1),
    log2_G007.LK = log2(G007.LK +1)
  )
human_data_transform1
```

This is the distribution of our data before the transformation

```{r}
human_data_transform1 |> 
  ggplot(mapping = aes(x = DMSO))+
  geom_histogram()
```

```{r}
human_data_transform1 |> 
  ggplot(mapping = aes(x = G007.LK))+
  geom_histogram()
```

These are the distributions after the transformation

```{r}
human_data_transform1 |> 
  ggplot(mapping = aes(x = log2_DMSO))+
  geom_histogram()
```

We can see here that we reduced some of the skewness in our data

```{r}
human_data_transform1 |> 
  ggplot(mapping = aes(x = log2_G007.LK))+
  geom_histogram()
```

\##############################################

Approach 2 We can use an the minimum positive value contained in the dataset as an offset based on the dataset characterisitics.

```{r}
human_data_trans_long |> 
  filter(Expression_levels > 0) |> 
  arrange(Expression_levels) |> 
  select(treatment, Expression_levels)
```

```{r}
#min positive value
min_pos_value<- min(human_data_trans_long$Expression_levels> 0)# I selected this feature as the one with the min value because I found that 

human_data_transform2<-
  human_data_trans_long |> 
  mutate(log2_Expression_levels = log2(Expression_levels + min_pos_value)
         )

human_data_transform2
```

```{r}
human_data_transform2 |> 
  ggplot(mapping = aes(
    x = log2_Expression_levels
  ))+
  geom_freqpoly()+
  scale_x_continuous(breaks = seq(-30,15, by = 5))
  
  

```

Here we can see that the majority of the observations are gathered between -5 +5.

Which reveals may indicate a binomial distribution. We have to look more into it.

\##############################################################################################################

## Do not pay attention to what is written below. These are just tests

\########################################################################################################

We are going to perform some tests in order to evaluate the significance of the experiment;s results

t-test : T- test is a type of statistical anaylsis used in order to compare the averages of two groups and determine whether differences between them are significant or not.

So we are going to use this test in order to assess the significancy of the results of the experiment.

The null hypothesis is that there are no differentially expressed genes. So we are going to perform some tests in order to see if we can reject it or not.

The first test is a paired t-test ince we want to estimate the p-values of the p-values before and after the treatment simultaneously

```{r}
human_data_transform1_tests<- human_data_transform1 |> 
  rowwise() |> #we use this function because we want to apply t.test to every one of our observations and get the p.value of each observation
  mutate(p_values = 
    t.test(c(log2_G007.LK, log2_DMSO, paired = TRUE))$p.value, #this extracts the pvalues of the ttest (t.test(c(treatment, control))$p.value) those are the raw p_values
    q_values = p.adjust(p_values))
human_data_transform1_tests
```

\###############################################

We are going to estimate the p-value of the t-statis

```{r}
human_data_transform1_tests |> 
  ggplot(mapping = aes(x = p_values)) +
  geom_freqpoly()
```

We know that the q-value is a better estimate to calculate multiple

DeSeq2

This distribution of the t-statistic suggests that on average there is not a strong global effect between the two compounds.

The bulk of t-statistics near 0 suggests that, for most genes, there is little to no difference in expression between the two groups. This is a common observation, as many genes are not expected to respond strongly to experimental conditionsHowever we can witness a second peak close to 0.5 is interesting and may indicate some meaningful biological phenomena

The peak around 0.5 in the tail could indicate that there are some genes with moderate but consistent differences in the expression between the two conditions. We could potentially perform a more thorough analysis in order to identify them. Caution this tail could by caused by biological Heterogeneity.

We are going to perform an extra test using the q-value of our results

```{r}
human_joined_aug <- human_joined_aug |> 
  mutate(
    log2fold_expression = log2(G007.LK/ DMSO)
  )
human_data_aug
```
