---
title: "Describe file"
format: html
editor: visual
---

```{r}
library(tidyverse)
library(dplyr)
library(ggplot2)
library(tidyr)
library(patchwork)

```

```{r}
#load the clean datasets that we created in the clean.qmd file
human_data_clean$DMSO<- as.numeric(human_data_clean$DMSO)
human_data_clean$G007.LK<-as.numeric(human_data_clean$G007.LK)
```

Categorical variables: gene_id, symbol, Cell_line, Compound, Organism, Genotype, Factor_unit_dose

Numerical variables: Expression_levels (Measured in Transcripts per million)

### Before performing the descriptive statistics we need to log2 transform the Expression levels in order to reduce the heteroscedasticity and stabilize the variance among the expression levels. ASK THE GUYS IF WE NEED TO DO THIS

```{r}
human_log_trans<- human_data_clean|> 
  mutate(
    log2_DMSO_Expression_levels = log2(DMSO +1),
    log2_G007.LK_Expression_levels = log2(G007.LK+1)
  )
human_log_trans
```

```{r}
#create a histogram
human_log_trans|> 
  ggplot(mapping = aes(x = log2_DMSO_Expression_levels))+
           geom_freqpoly(aes(color = Cell_line))+
  labs(x= "Expression Levels",
       y = "Density",
       title = "Density of the expression of human skin cells in DMSO ")

```

We see that there are many 0 but how many are there?

```{r}

#create a new variable called Expression_differences
human_log_trans |> 
  filter(log2_DMSO_Expression_levels == 0) |> 
  count(n = n())
```

How many are expressed though?

```{r}
human_log_trans |> 
  filter(log2_DMSO_Expression_levels != 0) |> 
  count(n = n())
```

What happens when we add G007,LK

```{r}
human_log_trans|> 
  ggplot(mapping = aes(x = log2_G007.LK_Expression_levels))+
           geom_freqpoly(mapping = aes(color = Cell_line))+
  labs(x= "Expression Levels",
       y = "Density",
       title = "Density of the expression of human skin cells when introducing G007.LK")
```

```{r}
human_log_trans |> 
  filter(log2_G007.LK_Expression_levels == 0) |> 
  count(n = n())
```

These are the genes that are not expressed by the DMSO

From the last two illustrations we can see that half of the genes are not expressed when DMSO is introduced

We can see that there are less genes that there are 2731 more genes that were expressed when we introduced the G007.LK

```{r}
#|label: install dependencies
install.packages(c("BiocManager", "BiocParallel"))
```

Estimate the difference in gene expression when we add the G007.LK. DESeq analysis..

```{r}

if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("DESeq2")
```

Questions to be investigated:

What type of variation occurs within the "Expression_levels Variable"?

Covariation between the compound and the Expression Levels?

Is that relationship affected by other variables?

Perform DESeq2 analysis to estimate the differences in expression levels!
