---
title: "Describe file"
format: html
editor: visual
---

```{r}
library(tidyverse)
library(dplyr)
library(ggplot2)
library(tidyr)
library(patchwork)
```

Categorical variables: Gene, Cell_line, Treatment, Genotype

Numerical variables: Expression_levels (Measured in transcripts per million)

We check the distribution of expression levels

###DESC_1

```{r}
###DESC_1 
DMSO_freq <- human_joined |> 
  filter(treatment == "DMSO") |> 
  ggplot(mapping = aes(x = expression_levels)) +
  geom_freqpoly()

G007.LK_freq <- human_joined |>
  filter(treatment == "G007.LK") |> 
  ggplot(mapping = aes(x = expression_levels)) +
  geom_freqpoly()

DMSO_freq + G007.LK_freq
```

We observe many observation equal to zero, but how many of them are there for each treatment?

###DESC_2

```{r}
#DESC_2 
human_joined |> 
  filter(Treatment == "DMSO") |> 
  summarise(DMSO_0 = sum(Expression_levels == 0))

human_joined |>
  filter(Treatment == "G007.LK") |> 
  summarise(G007.LK_0 = sum(Expression_levels == 0))
```

We then check the distribution of expression levels after log2 transformation

```{r}
### DESC_3
 DMSO_freq_log <- human_trans |>
   ggplot(mapping = aes(x = log2DMSO))+
   geom_freqpoly() #

 G007.LK_freq_log <- human_trans |>
   ggplot(mapping = aes(x = log2G007.LK))+
   geom_freqpoly()


 DMSO_freq_log + G007.LK_freq_log
```

We can see now that the distribution is less skewed but we can still see that the majority of the observations are equal to zero (equal to -33.92 after the log2 transformation).

The bulk of zeros suggests that, for many genes in many cell lines, there is no difference in expression between the two groups. This is a common observation, as many genes in different cell lines may not respond strongly to experimental conditions

```{r}
human_diff_expr |> 
  sample_n(100) |>
  ggplot(aes(x = Cell_line, 
             y = Gene, 
             fill = log2fold))+
  geom_tile()+
  scale_fill_gradientn(colors=c('blue','white','salmon1'))+
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1),axis.text=element_text(size=4),axis.title=element_text(size=5))+
  theme(legend.position = 'bottom',legend.key.size = unit(5, 'mm'),legend.title= element_text(size=5),legend.text=element_text(size=5))+
  coord_fixed(ratio=0.6)
```

Here we can see that teh distribution of both of the compounds is 0. This is something that is anticipated since we are dealing with RNA data and not all of the genes are expressed in all cell types.

In order to move on with our analysis we have to log transform our data so we can get some insghts

What happens when we add G007,LK

```         
```

These are the genes that are not expressed by the DMSO

From the last two illustrations we can see that half of the genes are not expressed when DMSO is introduced

We can see that there are less genes that there are 2731 more genes that were expressed when we introduced the G007.LK

Estimate the difference in gene expression when we add the G007.LK. DESeq analysis..

Questions to be investigated:

What type of variation occurs within the "Expression_levels Variable"?

Covariation between the compound and the Expression Levels?

Is that relationship affected by other variables?

Perform DESeq2 analysis to estimate the differences in expression levels!
